src/crimson/common/interruptible_future.h

```diff
@@ -256,20 +255,19 @@ Result non_futurized_call_with_interruption(
   InterruptCondRef<InterruptCond> interrupt_condition,
   Func&& func, T&&... args)
 {
-  if (interrupt_condition) {
-    auto [interrupt, fut] = interrupt_condition->template may_interrupt<seastar::future<>>();
-    INTR_FUT_DEBUG(
-      "non_futurized_call_with_interruption may_interrupt: {}, "
-      "interrupt_condition: {}, interrupt_cond: {},{}",
-      interrupt,
-      (void*)interrupt_condition.get(),
-      (void*)interrupt_cond<InterruptCond>.interrupt_cond.get(),
-      typeid(InterruptCond).name());
-    if (interrupt) {
-      std::rethrow_exception(fut->get_exception());
-    }
-    interrupt_cond<InterruptCond>.set(interrupt_condition);
-  }
+  assert(interrupt_condition);
+  auto [interrupt, fut] = interrupt_condition->template may_interrupt<seastar::future<>>();
+  INTR_FUT_DEBUG(
+    "non_futurized_call_with_interruption may_interrupt: {}, "
+    "interrupt_condition: {}, interrupt_cond: {},{}",
+    interrupt,
+    (void*)interrupt_condition.get(),
+    (void*)interrupt_cond<InterruptCond>.interrupt_cond.get(),
+    typeid(InterruptCond).name());
+  if (interrupt) {
+    std::rethrow_exception(fut->get_exception());
+  }
+  interrupt_cond<InterruptCond>.set(interrupt_condition);
```

Comment:
is it reasonable to merge this commit to 85c8915?

Reply:
I'm not quite sure, I thought this commit is relative independent